celery/utils/functional.py

```diff
@@ -242,11 +242,20 @@ def __bool__(self):
     @property
     def data(self):
         try:
-            self.__consumed.extend(list(self.__it))
+            if not self.__done:
+                self.__consumed.extend(list(self.__it))
+            self.__done = True
```

Comment:
Might be movedunder `if`
```suggestion
                self.__done = True
```

Reply:
The newly pushed diff makes this change due to the dedent of the rest of the try body